### ISSUE TITLE
Malloc Block memory leaks

### ISSUE THREAD

**Issue by Togishiro** [Swinject/Swinject#359]
Hello. When I checked the project for memory leaks, I noticed that each call resolver.resolve leads to memory accounting

![screen shot 2018-06-28 at 15 40 26](https://user-images.githubusercontent.com/15949410/42034760-b428be66-7ae9-11e8-9e00-6e0d778bf4ef.png)

Additionally, some calls container.register also lead to leaks
![screen shot 2018-06-28 at 15 47 11](https://user-images.githubusercontent.com/15949410/42035040-96998a46-7aea-11e8-91dc-5a1275998e79.png)

using Swinject version 2.4.1

**Comment by jakubvano** [comment]
This looks like the same issue as #352

**Comment by Togishiro** [comment]
you're right, but in this issue does not describe the leak at registration.
this leak occurs when casting to AnyObject.

**Comment by jakubvano** [comment]
Do you have some code which can reproduce this? It might help with investigation into why it is happening.

**Comment by Togishiro** [comment]
I can not send the code, because the project is commercial. But the bug occurs with this code structure

func a(_ container: SwinjectContainer) {
  configure1()
  configure2()
}

fjunc configure1() {
  container.register 1
  container.register 2
  container.register 3
}

func configure2()) {
  container.register 4
  container.register 5
  container.register 6
}

but when I wrote, there were no memory leaks
func b(_ container: SwinjectContainer) {
  configure2()
  configure1()  
}

memory leak occurs in file Container.swift, in this string
![screen shot 2018-06-29 at 09 28 08](https://user-images.githubusercontent.com/15949410/42076687-d5557ada-7b7e-11e8-98b3-ccd5a781cf11.png)

**Comment by Togishiro** [comment]
the leak generated at the container.resolve does not occur if the object is the NSObject heir

**Comment by rinat-enikeev** [comment]
Same issue for me. I have read https://github.com/Swinject/Swinject/issues/352. Any news regarding this bug?

**Comment by marekpridal** [comment]
Hi @jakubvano is there a chance to be mem leak bugs fixed in next version? #373 #359 #352 Thanks a lot 🙌

**Comment by jakubvano** [comment]
> the leak generated at the container.resolve does not occur if the object is the NSObject subclass

This suggests that the issue is the same as #352 - false positive report in memory analyser. It is difficult to telll without reproducible example though.

**Comment by jakubvano** [comment]
@marekpridal Based on current info #352 & #359 do not seem like a Swinejct bugs. As for #373 I probably wont be able to tackle it in the neer future - you are welcome to try though, it should be quite straightforward ;)
